UR20 description and meshes (#657)
ros-industrial/universal_robot: README.md

[![License - apache 2.0](https://img.shields.io/:license-Apache%202.0-yellowgreen.svg)](https://opensource.org/licenses/Apache-2.0)
[![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)

[![License](https://img.shields.io/badge/License-Universal%20Robots%20A/S%E2%80%99%20Terms%20and%20Conditions%20for%20Use%20of%20Graphical%20Documentation-blue.svg)](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)\

[![support level: community](https://img.shields.io/badge/support%20level-community-lightgray.png)](http://rosindustrial.org/news/2016/10/7/better-supporting-a-growing-ros-industrial-software-platform)

[![support level: community](https://img.shields.io/badge/support%20level-community-lightgray.png)](http://rosindustrial.org/news/2016/10/7/better-supporting-a-growing-ros-industrial-software-platform)

[ROS-Industrial](https://wiki.ros.org/Industrial) Universal Robot meta-package. See the [ROS wiki](https://wiki.ros.org/universal_robots) page for compatibility information and other more information.

> __License__

Reviewer: This is missing an empty line. Sorry, I didn't catch this in my suggestion.
```suggestion
__License__

```

![image](https://github.com/ros-industrial/universal_robot/assets/491645/544eba1b-35c9-4f6a-8a98-d657a5a18acc)

[![Build Status: Ubuntu Bionic (Actions)](https://github.com/ros-industrial/universal_robot/workflows/CI%20-%20Ubuntu%20Bionic/badge.svg?branch=melodic-devel)](https://github.com/ros-industrial/universal_robot/actions?query=workflow%3A%22CI+-+Ubuntu+Bionic%22)
[![Build Status: Ubuntu Focal (Actions)](https://github.com/ros-industrial/universal_robot/workflows/CI%20-%20Ubuntu%20Focal/badge.svg?branch=melodic-devel)](https://github.com/ros-industrial/universal_robot/actions?query=workflow%3A%22CI+-+Ubuntu+Focal%22)

[![License - apache 2.0](https://img.shields.io/:license-Apache%202.0-yellowgreen.svg)](https://opensource.org/licenses/Apache-2.0)
[![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)

> [![License](https://img.shields.io/badge/License-Universal%20Robots%20A/S%E2%80%99%20Terms%20and%20Conditions%20for%20Use%20of%20Graphical%20Documentation-blue.svg)](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)

Reviewer 2: Should we instead make this point to the `ur_description/meshes/ur20/LICENSE.txt` file? The online version could be changed at any time. The files in this repository would fall under the version of the license as it is right now, in this PR. It would make sense to me to make this badge link to the version the files would fall under.

Reviewer: From what I understood, the license is structured such that the latest version always is relevant (see point 1.4)

> ... If you download a software package that includes the Graphical Documentation, your use of the Graphical Documentation will be subject to and governed by the latest version of the T&Cs, which can be found here: ...

Given that clause I think it does make sense to link to the online version.

Reviewer 2: I'm not sure that's legal. There would be no change notification to current users, UR would be able to silently, unilaterally change the complete license without anyone being aware of it, and with all licenses I've worked with (and in the jurisdictions I've had to work with licenses), once you have a copy of the artefacts under that license, that copy and/or version will always fall under the version of the license as it was when you agreed to it.

I have seen licenses where there are provisions for updates made by the licensor, but those always include some sort of notification period, allowing current licensees to terminate their use of whatever is under the existing license if they feel they can't, or don't want to, agree to the new version.

Additionally, there is already a copy of the current version in the repository -- or at least there will be, after merging this PR.

It's best practice to include a copy of the license in all cases, even with OSS licenses. We don't do that currently in this repository, but that's an omission and should be fixed (in a later PR).

Reviewer 2: The same section 1.4 also states:
> However, you may only make the Graphical Documentation or any whole or partial copies thereof available to the public via download if it is part of a downloadable software package that includes a copy of the T&Cs.

which is what this PR will result in: a copy will be included.

So it doesn't seem strange to me to then point to that copy, and not the online version.

That seems like it would just lead to confusing situations.

Author: You have a point. Though the motivation was to remove the burden from the repo owner to ensure that the license is up to date.

> [![License](https://img.shields.io/badge/License-Universal%20Robots%20A/S%E2%80%99%20Terms%20and%20Conditions%20for%20Use%20of%20Graphical%20Documentation-blue.svg)](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)\

Reviewer: ```suggestion
[![License](https://img.shields.io/badge/License-Universal%20Robots%20A/S%E2%80%99%20Terms%20and%20Conditions%20for%20Use%20of%20Graphical%20Documentation-blue.svg)](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)
```

> [![support level: community](https://img.shields.io/badge/support%20level-community-lightgray.png)](http://rosindustrial.org/news/2016/10/7/better-supporting-a-growing-ros-industrial-software-platform)
> 

Reviewer 2: was this duplication intentional?

```suggestion
```

[![support level: community](https://img.shields.io/badge/support%20level-community-lightgray.png)](http://rosindustrial.org/news/2016/10/7/better-supporting-a-growing-ros-industrial-software-platform)

[ROS-Industrial](https://wiki.ros.org/Industrial) Universal Robot meta-package. See the [ROS wiki](https://wiki.ros.org/universal_robots) page for compatibility information and other more information.

__License__

> The [UR20 meshes](ur_description/meshes/ur20) constitutes “Graphical Documentation” the use of which is subject to and governed by our “[Terms and Conditions for use of Graphical Documentation](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)”.\

Reviewer 2: ```suggestion
Note: the mesh files used by the [UR20](ur_description/meshes/ur20) model in the `ur_description` package come with a non-OSI approved, non-open-source license.
They are considered *Graphical Documentation* by Universal Robots and their use is subject to and governed by the "[Terms and Conditions for use of Graphical Documentation](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)".

Users of these packages are advised to make sure their use complies with these additional terms and conditions.
Please read the license carefully before making use of the UR20 variant in the `ur_description` package.
```

Edit: I would actually suggest to make the link here *also* point to `ur_description/meshes/ur20/LICENSE.txt`, just as the badge (see earlier comment).

I don't know what the link would become though, so I don't have a suggestion to easily accept and apply I'm afraid.

Reviewer: @urrsk as far as I remember this section is an outcome of UR's law department, so I guess I cannot say much to that.

Reviewer 2: The original section contained style violations and typos. Additionally, we can't have it state "our [license]", as this is a repository on the `ros-industrial` organisation. It's not owned by UR, and we are not UR.

The rest of the changes merely draw attention to the fact that legally, users should *not* use any of the files (and perhaps not even download them) under the new license until after they've determined they can and will agree to its terms.

I'm sure lawyers will be able to find fault with just about anything, but I can't really see what's wrong here. It's definitely a non-OSI approved, non-OSS license. That's just a fact, and I feel important to call out.

Reviewer 2: @gavanderhoorn wrote:
> The rest of the changes merely draw attention to the fact that legally, users should _not_ use any of the files (and perhaps not even download them) under the new license until after they've determined they can and will agree to its terms.

which is exactly what the new license also states/asks (from the preamble of `v1.01`):
> The T&Cs constitute a binding agreement between you and Universal Robots. By accessing, storing, copying, sharing, opening, or otherwise using or disposing of the Graphical Documentation, you acknowledge that you are bound by the T&Cs.

Author: @gavanderhoorn thanks for your feedback.
Agree, the wording is made for a repo UR's Github organisation.
I will come back with another suggestion.
And yes it is not a OSI approved license

Author: Here is my suggestion.
It is important to let people understand that it is a proprietary license, though I also would like to highlight that it pretty open as well.
```suggestion
The [UR20 meshes](ur_description/meshes/ur20) constitute “Graphical Documentation” from Universal Robots which is subject to and governed by Universal Robots' "[Terms and Conditions for use of Graphical Documentation](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)".

Universal Robots' [Terms and Conditions for use of Graphical Documentation](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt) do not fully comply with [OSI's definition of Open Source](https://opensource.org/osd/), but they do allow you to use, modify and share “Graphical Documentation”, including [UR20 meshes](ur_description/meshes/ur20), subject to certain restrictions.\
```

> If you have any questions regarding the license or the license doesn't fit you use-case, please contact [legal@universal-robots.com](mailto:legal@universal-robots.com).

Reviewer 2: ```suggestion
If you have any questions regarding this license or if this license doesn't fit your use-case, please contact [legal@universal-robots.com](mailto:legal@universal-robots.com).
```

[![Build Status: Ubuntu Bionic (Actions)](https://github.com/ros-industrial/universal_robot/workflows/CI%20-%20Ubuntu%20Bionic/badge.svg?branch=melodic-devel)](https://github.com/ros-industrial/universal_robot/actions?query=workflow%3A%22CI+-+Ubuntu+Bionic%22)
[![Build Status: Ubuntu Focal (Actions)](https://github.com/ros-industrial/universal_robot/workflows/CI%20-%20Ubuntu%20Focal/badge.svg?branch=melodic-devel)](https://github.com/ros-industrial/universal_robot/actions?query=workflow%3A%22CI+-+Ubuntu+Focal%22)

[![License - apache 2.0](https://img.shields.io/:license-Apache%202.0-yellowgreen.svg)](https://opensource.org/licenses/Apache-2.0)
[![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)

> The [UR20 meshes](ur_description/meshes/ur20) constitutes “Graphical Documentation” the use of which is subject to and governed by our “Terms and Conditions for use of Graphical Documentation”
> [![License](https://img.shields.io/badge/License-Universal%20Robots%20A/S%E2%80%99%20Terms%20and%20Conditions%20for%20Use%20of%20Graphical%20Documentation-blue.svg)](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)\
> If you have any questions regarding the license or the license doesn't fit you use-case, please contact [legal@universal-robots.com](mailto:legal@universal-robots.com)
> 
> [![support level: community](https://img.shields.io/badge/support%20level-community-lightgray.png)](http://rosindustrial.org/news/2016/10/7/better-supporting-a-growing-ros-industrial-software-platform)
> 
> [ROS-Industrial](https://wiki.ros.org/Industrial) Universal Robot meta-package. See the [ROS wiki](https://wiki.ros.org/universal_robots) page for compatibility information and other more information.

Reviewer: I think it makes sense to separate the license badge and the statement the following way:
```suggestion
[![License](https://img.shields.io/badge/License-Universal%20Robots%20A/S%E2%80%99%20Terms%20and%20Conditions%20for%20Use%20of%20Graphical%20Documentation-blue.svg)](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)\

[![support level: community](https://img.shields.io/badge/support%20level-community-lightgray.png)](http://rosindustrial.org/news/2016/10/7/better-supporting-a-growing-ros-industrial-software-platform)

[![support level: community](https://img.shields.io/badge/support%20level-community-lightgray.png)](http://rosindustrial.org/news/2016/10/7/better-supporting-a-growing-ros-industrial-software-platform)

[ROS-Industrial](https://wiki.ros.org/Industrial) Universal Robot meta-package. See the [ROS wiki](https://wiki.ros.org/universal_robots) page for compatibility information and other more information.

__License__
The [UR20 meshes](ur_description/meshes/ur20) constitutes “Graphical Documentation” the use of which is subject to and governed by our “[Terms and Conditions for use of Graphical Documentation](https://www.universal-robots.com/legal/terms-and-conditions/terms_and_conditions_for_use_of_graphical_documentation.txt)”.\
If you have any questions regarding the license or the license doesn't fit you use-case, please contact [legal@universal-robots.com](mailto:legal@universal-robots.com).

All other content is licensed under the BSD-3-Clause license or Apache-2.0 respectively.

```

Author: Good suggestion. Lets do that
